### PR TITLE
WIP: Try to send a grpc request from TypeScript to Go

### PR DESCRIPTION
**Pull request by anxiousmodernman**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -369,12 +370,18 @@ func serve(conf config.Config) error {
...
+	)
+
 	// All websockets requests are POSTs to some {method} on this route.
 	p.Handle("/web.Proxy/{method}", negroni.New(
 		setConf(conf),
 		negroni.HandlerFunc(withLog),
 		negroni.HandlerFunc(authHandler),
-		negroni.Wrap(websocketsProxy(wsproxy)),
+		negroni.Wrap(allowCORS(websocketsProxy(wsproxy))),
 	)).Methods("POST")
```
Comment (anxiousmodernman) [inline main.go]: All grpc requests go to `/web.Proxy/{methodName}`. Do we need to explicitly add OPTIONS handling somewhere in here?

File: main.go [hunk 2]
```diff
@@ -415,8 +422,9 @@ func serve(conf config.Config) error {
 
 	// Web server for our Vecty/GopherJS management UI
 	httpsSrv := &http.Server{
-		Addr:              fmt.Sprintf("0.0.0.0:%s", conf.WebUIPort),
-		Handler:           p,
+		Addr: fmt.Sprintf("0.0.0.0:%s", conf.WebUIPort),
+		// TODO fix me, make me configurable based on local/prod builds
+		Handler:           handlers.CORS(handlers.AllowedOrigins([]string{"*"}))(p),
```
Comment (anxiousmodernman) [inline main.go]: Redundant wrapping here?

File: proto/web.proto [hunk 3]
```diff
@@ -2,10 +2,11 @@ syntax = "proto3";
 
 package web;
 
-import "github.com/johanbrandhorst/protobuf/proto/gopherjs.proto";
+//import "github.com/johanbrandhorst/protobuf/proto/gopherjs.proto";
```
Comment (anxiousmodernman) [inline proto/web.proto]: We remove gopherjs code gen instructions from our protobuf

File: ui/build.sh [hunk 4]
```diff
@@ -2,17 +2,28 @@
...
+# don't need this???
+#tsc
 
 # move CSS assets
-cp src/*.css build/dist
+#cp src/*.css build/dist
 
 # run parcel build
-parcel index.html
+#parcel index.html
```
Comment (anxiousmodernman) [inline ui/build.sh]: run `parcel index.html` manually after `./build.sh` until we sort this out.

File: ui/generated/web_pb.d.ts [hunk 5]
```diff
@@ -0,0 +1,216 @@
+// package: web
```
Comment (anxiousmodernman) [inline ui/generated/web_pb.d.ts]: We check in generated typescript.

File: ui/src/App.tsx [hunk 6]
```diff
@@ -1,16 +1,42 @@
 import * as React from 'react';
+import { Backend, StateRequest, ProxyState } from '../generated/web_pb';
+import { ProxyClient, ServiceClientOptions } from '../generated/web_pb_service';
 import './App.css';
 
 // import logo from './logo.svg';
 
+
```
Comment (anxiousmodernman) [inline ui/src/App.tsx]: Here we make use of our generated client and request objects. We call the "state" method, which reports the proxy's state... when it works.

**Comment by Nearhan** [comment]
https://github.com/improbable-eng/grpc-web/blob/master/ts/docs/client.md

we probably need to do this if we want to test the lib

**Pull request merged**